includes/payment-methods/class-wc-stripe-upe-payment-method-amazon-pay.php

```diff
@@ -24,6 +25,36 @@ public function __construct() {
 			'Amazon Pay is a payment method that allows customers to pay with their Amazon account.',
 			'woocommerce-gateway-stripe'
 		);
+		$this->supports[]           = 'tokenization';
+
+		// Check if subscriptions are enabled and add support for them.
+		$this->maybe_init_subscriptions();
+	}
+
+	/**
+	 * Returns string representing payment method type
+	 * to query to retrieve saved payment methods from Stripe.
+	 */
+	public function get_retrievable_type() {
+		return $this->get_id();
+	}
+
+	/**
+	 * Create new WC payment token and add to user.
+	 *
+	 * @param int $user_id        WP_User ID
+	 * @param object $payment_method Stripe payment method object
+	 *
+	 * @return WC_Payment_Token_Amazon_Pay
+	 */
+	public function create_payment_token_for_user( $user_id, $payment_method ) {
+		$token = new WC_Payment_Token_Amazon_Pay();
+		$token->set_email( $payment_method->billing_details->email ?? '' );
```

Comment:
Is there a scenario where a billing email will not be provided? Wondering if we should allow empty emails in the token or error out.

Reply:
Yeah, the check is most likely not needed, according to https://docs.stripe.com/api/payment_methods/object#payment_method_object-billing_details-email. I guess I thought it best to gracefully handle, as it's run at a point where the subscription is already paid for, and an error might make the shopper re-attempt payment.

What do you think?

Reply:
ah makes sense, let's keep it as is.